Kaguya/Kaguya/Discord/Parsers/UrlParser.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Text.RegularExpressions;
+
+namespace Kaguya.Discord.Parsers
+{
+    public class UrlParser
```

Comment:
why did you make your own url parser instead of using https://docs.microsoft.com/en-us/dotnet/api/system.uri?view=net-5.0

Reply:
Can this check if a string contains a url? That was my primary reason behind the regexes.

Reply:
i think it might be better to split on space and just check each segment

`str.Split(' ').Where(s=>Uri.TryCreate(s,UriKind.RelativeOrAbsolute,out var _)).Select(s=>new Uri(s))`